test/test-handlebars.js

```diff
@@ -730,6 +731,98 @@ test('reply.view with handlebars engine with partials', t => {
   })
 })
 
+test('reply.view with handlebars engine with missing partials path', t => {
+  t.plan(5)
+  const fastify = Fastify()
+  const handlebars = require('handlebars')
+  const data = { text: 'text' }
+
+  fastify.register(require('../index'), {
+    engine: {
+      handlebars: handlebars
+    },
+    options: {
+      partials: { body: './non-existent' }
+    }
+  })
+
+  fastify.get('/', (req, reply) => {
+    reply.view('./templates/index-with-partials.hbs', data)
+  })
+
+  fastify.listen(0, err => {
+    t.error(err)
+    sget({
+      method: 'GET',
+      url: 'http://localhost:' + fastify.server.address().port
+    }, (err, response, replyBody) => {
+      t.error(err)
+      t.strictEqual(response.statusCode, 500)
+      t.strictEqual(response.headers['content-length'], '' + replyBody.length)
+      t.strictEqual(response.headers['content-type'], 'application/json; charset=utf-8')
+
+      fastify.close()
+    })
+  })
+})
+
+test('reply.view with handlebars engine with partials in production', t => {
+  t.plan(6)
+  const fastify = Fastify()
+  const handlebars = require('handlebars')
+  const data = { text: 'text' }
+
+  fastify.register(require('../index'), {
+    engine: {
+      handlebars: handlebars
+    },
+    options: {
+      partials: { body: './templates/body.hbs' }
+    },
+    production: true
+  })
+
+  fastify.get('/', (req, reply) => {
+    reply.view('./templates/index-with-partials.hbs', data)
+  })
+
+  fastify.listen(0, err => {
+    t.error(err)
+    sget({
+      method: 'GET',
+      url: 'http://localhost:' + fastify.server.address().port
+    }, (err, response, replyBody) => {
+      t.error(err)
+      t.strictEqual(response.statusCode, 200)
+      t.strictEqual(response.headers['content-length'], '' + replyBody.length)
```

Comment:
```suggestion
      t.strictEqual(response.headers['content-length'], String(replyBody.length))
```

Reply:
This pattern is in multiple tests in this file should I change it in all of them ?

Reply:
No, it is not necessary, I saw the `require()` too could be moved at the top but it is already done in other tests, but it would add to much noise to the PR.